libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -166,7 +192,54 @@ func (k *kubernetesAnnotator) Run(event *beat.Event) (*beat.Event, error) {
 	return event, nil
 }
 
-func (*kubernetesAnnotator) String() string { return "add_kubernetes_metadata" }
+// worker watches pod events and keeps a map of metadata
+func (k *kubernetesAnnotator) worker() {
```

Comment:
i like this a lot. lets change the metricbeat event module to use this as well and rely on the same watcher implementation as in this PR.

Reply:
SGTM, let's wait for the outcome of #5949 as I think it may help here